internal/worker/worker.go

```diff
@@ -201,6 +203,46 @@ func (worker *Worker) syncVMs(ctx context.Context) error {
 		}
 	}
 
+	// Sync on-disk VMs
+	if err := worker.syncOnDiskVMs(ctx, remoteVMs); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (worker *Worker) syncOnDiskVMs(ctx context.Context, remoteVMs map[string]v1.VM) error {
+	onDiskNamesRaw, err := tart.List(ctx, worker.logger)
+	if err != nil {
+		return err
+	}
+
+	for _, onDiskNameRaw := range onDiskNamesRaw {
+		onDiskName, err := ondiskname.Parse(onDiskNameRaw)
+		if err != nil {
+			continue
+		}
+
+		remoteVM, ok := remoteVMs[onDiskName.UID]
+		if !ok {
+			// On-disk VM doesn't exist on the controller, delete it
+			_, _, err := tart.Tart(ctx, worker.logger, "delete", onDiskNameRaw)
+			if err != nil {
+				return err
+			}
+		} else if !worker.vmm.Exists(v1.VM{UID: onDiskName.UID}) {
```

Comment:
This should fail a remote VM only if it's in a Running state. If a VM was stopped and a worker restarts it's fine to have it as stopped.

Reply:
Good catch, fixed in 63bebd1.